### PR TITLE
Update usaco-858.mdx

### PR DESCRIPTION
**Pull request by ascyrax**
Added support for C++ viz Added a working C++ implementation

### CODE REVIEW

File: solutions/usaco-858.mdx [hunk 1]
```diff
@@ -136,4 +136,98 @@ public class convention {
...
 
 </JavaSection>
+
+
+<CPPSection>
+
+```cpp
+
+// @author: ascyrax
+
```
Comment (SansPapyrus683) [inline solutions/usaco-858.mdx:146]:
```suggestion
```
instead, add your name to the Authors field above

File: solutions/usaco-858.mdx [hunk 2]
```diff
@@ -136,4 +136,98 @@ public class convention {
...
+#include <bits/stdc++.h>
+using namespace std;
+
+#define endl "\n"
+
```
Comment (SansPapyrus683) [inline solutions/usaco-858.mdx:151]:
```suggestion
```
this macro is unapproved
don't think this would tle without it anyways

File: solutions/usaco-858.mdx [hunk 3]
```diff
@@ -136,4 +136,98 @@ public class convention {
...
+                int mid = l + (r - l) / 2;
+                if (check(mid))
+                        r = mid;
+                else
+                        l = mid + 1;
+        }
+
+        cout << l << endl;
+
+        return 0;
```
Comment (SansPapyrus683) [inline solutions/usaco-858.mdx:225]:
```suggestion
```

File: solutions/usaco-858.mdx [hunk 4]
```diff
@@ -136,4 +136,98 @@ public class convention {
...
+#define endl "\n"
+
+#define pb push_back
+
+int n, m, c;
+vector<int> vTime;
+
+bool check(int mid)
+{
+        //POINTERS for the cows to be added in the CURRENT BUS
```
Comment (SansPapyrus683) [inline solutions/usaco-858.mdx:159]: add a space after the `//` for comments
like
```suggestion
        // POINTERS for the cows to be added in the CURRENT BUS
```

File: solutions/usaco-858.mdx [hunk 5]
```diff
@@ -136,4 +136,98 @@ public class convention {
...
+// @author: ascyrax
+
+#include <bits/stdc++.h>
+using namespace std;
+
+#define endl "\n"
+
+#define pb push_back
+
+int n, m, c;
```
Comment (SansPapyrus683) [inline solutions/usaco-858.mdx:154]: `n` and `m` can be made local to `main`

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-858.mdx (hunk 5 above)
Comment (ascyrax) [reply solutions/usaco-858.mdx]: done

**Comment by caoash** [comment]
Looks good.

File: solutions/usaco-858.mdx [hunk 6]
```diff
@@ -136,4 +136,86 @@ public class convention {
...
+                nBuses++;
+        //total no of buses cannot be greater than m
+        return nBuses <= m;
+}
+
+int main()
+{
+        freopen("convention.in", "r", stdin);
+        freopen("convention.out", "w", stdout);
+	int n,m;
```
Comment (SansPapyrus683) [inline solutions/usaco-858.mdx:196]: wait no
@caoash bruh
the indentation here is really weird
wait just realized
@ascyrax can you make another pr that fixes the indentation here and adds spaces between operators

Comment (ascyrax) [reply solutions/usaco-858.mdx:196]: oops. sure

Comment (ascyrax) [reply solutions/usaco-858.mdx:196]: @SansPapyrus683 I just gave the code a look and found out that we cannot make variable n and m local to main as I am using them inside the "check" function.

Comment (ascyrax) [reply solutions/usaco-858.mdx:196]: @SansPapyrus683  I am gonna open a new pr with all the necessary changes.

Comment (SansPapyrus683) [reply solutions/usaco-858.mdx:196]: oh
if you're using them in the `check` function then alright
just make sure to convert to tabs
